### ISSUE TITLE
bug involving escaped characters

### ISSUE THREAD

**Issue by sam0x17** [DanilaFe/pegasus#2]
It looks like it is currently not possible to escape control characters such as +, -, *,etc:

```
S = INTEGER;
INTEGER = SIGN_OP NUMBER | NUMBER;
NUMBER = "[0-9]+";
SIGN_OP = "\+" | "\-";
```
results in:
```
an error has occurred while compiling regular expressions: Invalid operator
```

Not sure why, but yes I have confirmed that the error is originating fromthe `SIGN_OP =` line.

btw, thx for everything

**Comment by sam0x17** [comment]
update: I tried adding "+", "-", and "*" to the ESCAPES list, but this did not fix it

**Comment by DanilaFe** [comment]
Thanks a lot for this issue. Two things come into play here: 
* The back slash needs to be escaped twice (once for the string, once for the regex), so `\\+`
* I only allowed characters escapable in C++, which is blatantly wrong since it doesn't allow the regex control characters (they need to be added to the escape list like you described)

If my understanding of the problem is correct, I will be able to push the fix in about an hour.

**Comment by sam0x17** [comment]
thanks!! yeah this reflects my suspicions

**Comment by DanilaFe** [comment]
This should be fixed in 5cd399b086b4. Thanks a lot for the report!

**Comment by sam0x17** [comment]
:+1:  thanks!!
